File: src/MaksIT.Models/LetsEncryptServer/ApiKeys/Search/SearchAPIKeyResponse.cs
using MaksIT.Core.Abstractions.Webapi;

namespace MaksIT.Models.LetsEncryptServer.ApiKeys.Search;

public class SearchAPIKeyResponse : ResponseModelBase {
  public Guid Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public string? Description { get; set; }
  public DateTime? ExpiresAt { get; set; }
  public DateTime? RevokedAt { get; set; }
}

File: src/MaksIT.WebUI/src/models/apiKey/search/SearchApiKeyResponse.ts
import type { ResponseModelBase } from '@maks-it.com/webui-contracts'

/** Mirrors <c>MaksIT.CertsUI.Models.APIKeys.Search.SearchAPIKeyResponse</c>. */
export interface SearchApiKeyResponse extends ResponseModelBase {
  id: string
  createdAt: string
  description?: string | null
  expiresAt?: string | null
  isGlobalAdmin: boolean
}
